### ISSUE TITLE
Identify how sync can be made singleton

### ISSUE THREAD

**Issue by mahalakshme** [avniproject/avni-client#1150]
As a user I want only one sync to be triggered in my app at a time, so that the app works in the designed way.

Currently we have disabled auto sync when manual sync is in progress. But manual sync can happen when auto sync is in progress as mentioned [here](https://github.com/avniproject/avni-client/issues/1132#issuecomment-1767792602) and another manual sync can be triggered when a manual sync is in progress as mentioned in the steps to reproduce [here](https://github.com/avniproject/avni-client/issues/1116). This results in duplicate of data, unexpected behavior and added server load.

#### Acceptance criteria:
- Come up with a strategy to make the sync singleton by using lockfile or state variable or by any other way.
- This strategy should be such that it should work, even when the process that has acquired the lock gets killed for some reason.

**Comment by 1t5j0y** [comment]
Parallel syncs analysis:
1. Cannot start a second manual sync in parallel. If the app is in memory while a manual sync is in progress, the manual sync continues. On restoring the app, another manual sync cannot be started because the progress bar is shown. If the app is removed from working memory during manual sync, the sync stops. On restoring the app and starting manual sync, a fresh sync starts.
2. Cannot initiate a second automatic sync when an automatic or manual sync is in progress. The existing strategy to schedule a dummy sync with the same job name has the desired effect of preventing a duplicate automatic sync from triggering. However, this has a side effect of always triggering an upload only sync immediately after a manual sync has completed due to us scheduling the automatic sync job again at the end of of manual sync. (which pushes nothing since a manual sync has just completed and the user has been prevented from performing any action on the app during the sync due to the display of the progress bar).
3. Can initiate a manual sync while an automatic sync is in progress as detailed in #1132 .

Hence the only scenario left to be handled is 3. There is existing state `SyncActions.syncing` to indicate whether a sync is currently in progress. This is currently not being set when the automatic sync is going on. We can change logic to set this to true even for automatic sync and use it to update the sync button UI and disable it (without interrupting the user) to indicate that an automatic sync is in progress which will take care of this scenario. Some additional handling to not show the progress bar for this scenario will also need to be handled.

**Comment by mahalakshme** [comment]
This this state, SyncActions.syncing will get reset when the app gets killed and we start it again right? just confirming.

**Comment by 1t5j0y** [comment]
correct.
